## Application Analysis 

This project intends to analyze Android and iOS application to understnad which apps are the most popular and present the best opportunities to attract users.

We will be mining data about current apps availaable in boest the App Store and in Google Play Store.


In [2]:
opened_file = open('AppleStore.csv')
from csv import reader
read_file = reader(opened_file)
as_data = list(read_file)

opened_file = open('googleplaystore.csv')
from csv import reader
read_file = reader(opened_file)
gps_data = list(read_file)


In [3]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

After opening and placing the data into lists, we can explore the datasets. Here are the first five rows along with totals:

In [4]:
explore_data(as_data, 1, 5, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 7198
Number of columns: 16


In [5]:
explore_data(gps_data, 1, 5, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 10842
Number of columns: 13


Here are the column headers:

For more detail about these data sets, you can review the descriptions for the Google Play Store here: [here](http://https://www.kaggle.com/lava18/google-play-store-apps) and the Apple Store here: [here](https://dq-content.s3.amazonaws.com/350/AppleStore.csv)

In [6]:
explore_data(as_data, 0, 1)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']




In [7]:
explore_data(gps_data, 0, 1)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']




Relevant columns for the Apple Store could include: 'track_name', 'price', 'user rating', and 'genre'

Google Play columns could include: 'App', 'Catergory', 'Rating', 'Installs', 'Price' and 'Genres'

## Data Cleaning (missing data/errors)

Google Play Store data set users have highlighted a possible error in row `10472`. Lets look at that row:

In [8]:
explore_data(gps_data, 10472, 10474)

['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']




Indeed, it looks like row `10473` (when including header) is missing a column. Let's delete that row now using the `del` function: 

In [9]:
del gps_data[10473]

Lets search the Apple Store for missing column data:

In [10]:
for row in as_data:
    headerlength = len(as_data[0])
    rowlength = len(row) 
    if rowlength != headerlength:
        print(row)
        print(as_data.index(row))

## Data Cleaning (duplicates)

Fortunately, we didn't find any missing Apple Store data

Now lets look for duplicate apps. We'll use a for loop to divide the data set into two lists: `duplicate_apps` and `unique_apps`

In [11]:
duplicate_gps = []
unique_gps = []

for app in gps_data:
    name = app[0]
    if name in unique_gps:
        duplicate_gps.append(name)
    else:
        unique_gps.append(name)
        
print("Number of duplicate apps: ", len(duplicate_gps))
print('\n')
print("Examples of duplicae apps: ", duplicate_gps[:10])

Number of duplicate apps:  1181


Examples of duplicae apps:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']


There are 1181 duplicate apps in the Google play data set. Let's look at a few in detail to see what, if any data is different in each:

In [12]:

for app in gps_data:
    name = app[0]
    if name == "Box":
        print(app)
        
for app in gps_data:
    name = app[0]
    if name == "Slack":
        print(app)
        
for app in gps_data:
    name = app[0]
    if name == "Google Ads":
        print(app)
        

['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Box', 'BUSINESS', '4.2', '159872', 'Varies with device', '10,000,000+', 'Free', '0', 'Everyone', 'Business', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Slack', 'BUSINESS', '4.4', '51507', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']
['Slack', 'BUSINESS', '4.4', '51507', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']
['Slack', 'BUSINESS', '4.4', '51510', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies w

It appears that `box` was exactly the same, but `Slack` and `Google ads` had one entry with a higher review count. Since that suggests the most recent data, lets elect to keep those.

Now, let's remove any rows that are dupilicates--keeping only the higest rated ones.

First we'll create an empty dictionary to store them

In [13]:
reviews_max = {}

for app in gps_data[1:]:
    name = app[0]
    n_reviews = float(app[3])
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
        

Let's double check the size of the `reviews_max` file and compare to the `gps_data` file minus the duplicates

In [14]:
print("Expected length: ", len(gps_data) - 1181)
print('Actual length: ', len(reviews_max))

Expected length:  9660
Actual length:  9659


We'll now loop through the `gps_data` file again and compare with the highest reivews found in `reviews_max`

We will also add a `already_added` loop for those duplicates that have multiple with the same high rating

In [15]:
android_clean = []
already_added = []

for app in gps_data[1:]:
    name = app[0]
    n_reviews = float(app[3])
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)

In [16]:
print(len(android_clean))

9659


## Data Cleaning (non-English apps)

Now that we've removed duplicates in both the `as_data` and the `gps_data` -- now called `android_clean` we can look to remove any non-English apps.

First we'll create a function that examines if a given character fits in the ASCII system. This is the American Standard Code for Information Interchange, and all english characters have a number between 0-127. `A`, for example is 65.

In [17]:
def is_english(string):
    for letter in string:
        if ord(letter) > 127:
            return False
    return True

print(is_english('Instragram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Instachat 😜'))

True
False
False


Our `is_english` function worked, but it could delete some otherwise appropriate apps, like `'Instachat 😜'`.

let's add code that only returns false if there is 3 or more non-english characters.

In [18]:
def is_english(string):
    non_ascii = 0
    
    for letter in string:
        if ord(letter) > 127:
            non_ascii += 1
            
    if non_ascii > 3:
        return False
    else:
        return True

print(is_english('Instragram'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Instachat 😜'))

True
False
True


Our edited function works pretty well now. Let's iterate through our `as_data` and `android_clean` and remove any non-english apps

In [19]:
as_english = []
gps_english = []

for app in android_clean[1:]:
    name = app[0]
    if is_english(name):
        gps_english.append(app)
        
for app in as_data[1:]:
    name = app[1]
    if is_english(name):
        as_english.append(app)
        
print(len(as_english))
print(len(gps_english))   

6183
9613


## Data Cleaning (isolating free apps)

So far we have 1) removed inaccurate data, 2) removed duplicate entries and 3) removed non-English apps.

To meet the intended goal, our last data cleaning step is isolating the free apps:

In [20]:
as_final = []
gps_final = []

for app in as_english:
    price = app[4]
    if price == '0.0':
        as_final.append(app)
    
for app in gps_english:
    price = app[7]
    if price == '0':
        gps_final.append(app)
    
print(len(as_final))
print(len(gps_final))

3222
8863


## Data Analysis Part 1:

Now that we have cleaned our data sets, we can begin to analyize them.

Our goal is understand which apps are the  most popular, or successful. Our team plans to first launch in the Google Play Store, and if demonstraed to be profitable, build and launch an Apple Store version. Ideally, our app will need to be successful in both markets.

Let's being by understanding the most common genre's in each market. In the Apple Store we'll build a frequency table of the `prime_genre` column [11], and for the Google Play Store we'll create a table for the `category` [1] and `genre` [9] columns:

In [21]:
# function for frequency table with percentages

def freq_table(data_set, index):
    table = {}
    total = 0
    
    for row in data_set:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
    #return table

    table_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage
        
    return table_percentages

In [22]:
as_genre = freq_table(as_final, 11)
print(len(as_genre))
print(as_genre)


23
{'Travel': 1.2414649286157666, 'Navigation': 0.186219739292365, 'Reference': 0.5586592178770949, 'Medical': 0.186219739292365, 'Business': 0.5276225946617008, 'Sports': 2.1415270018621975, 'Shopping': 2.60707635009311, 'Photo & Video': 4.9658597144630665, 'Music': 2.0484171322160147, 'Entertainment': 7.883302296710118, 'Social Networking': 3.2898820608317814, 'Food & Drink': 0.8069522036002483, 'Utilities': 2.5139664804469275, 'Finance': 1.1173184357541899, 'Book': 0.4345127250155183, 'Productivity': 1.7380509000620732, 'Catalogs': 0.12414649286157665, 'News': 1.3345747982619491, 'Weather': 0.8690254500310366, 'Games': 58.16263190564867, 'Health & Fitness': 2.0173805090006205, 'Education': 3.662321539416512, 'Lifestyle': 1.5828677839851024}


In [23]:
#create a function to sort the percentages in order

def display_table(data_set, index):
    table = freq_table(data_set, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
    
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ":", entry[0])

In [24]:
as_top_genre = display_table(as_final, 11)
print(as_top_genre)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665
None


In [25]:
gps_top_category = display_table(gps_final, 1)
print(gps_top_category)

FAMILY : 18.910075595170937
GAME : 9.725826469592688
TOOLS : 8.462146000225657
BUSINESS : 4.592124562789123
LIFESTYLE : 3.9038700214374367
PRODUCTIVITY : 3.8925871601038025
FINANCE : 3.7007785174320205
MEDICAL : 3.5315355974275078
SPORTS : 3.396141261423897
PERSONALIZATION : 3.317161232088458
COMMUNICATION : 3.2381812027530184
HEALTH_AND_FITNESS : 3.0802211440821394
PHOTOGRAPHY : 2.944826808078529
NEWS_AND_MAGAZINES : 2.798149610741284
SOCIAL : 2.6627552747376737
TRAVEL_AND_LOCAL : 2.335552296062281
SHOPPING : 2.245289405393208
BOOKS_AND_REFERENCE : 2.1437436533904997
DATING : 1.8616721200496444
VIDEO_PLAYERS : 1.7939749520478394
MAPS_AND_NAVIGATION : 1.399074805370642
FOOD_AND_DRINK : 1.241114746699763
EDUCATION : 1.1621347173643235
ENTERTAINMENT : 0.9590432133589079
LIBRARIES_AND_DEMO : 0.9364774906916393
AUTO_AND_VEHICLES : 0.9251946293580051
HOUSE_AND_HOME : 0.8236488773552973
WEATHER : 0.8010831546880289
EVENTS : 0.7108202640189552
PARENTING : 0.6544059573507841
ART_AND_DESIGN : 0

In [26]:
gps_top_genre = display_table(gps_final, 9)
print(gps_top_genre)

Tools : 8.450863138892023
Entertainment : 6.070179397495204
Education : 5.348076272142616
Business : 4.592124562789123
Productivity : 3.8925871601038025
Lifestyle : 3.8925871601038025
Finance : 3.7007785174320205
Medical : 3.5315355974275078
Sports : 3.463838429425702
Personalization : 3.317161232088458
Communication : 3.2381812027530184
Action : 3.102786866749408
Health & Fitness : 3.0802211440821394
Photography : 2.944826808078529
News & Magazines : 2.798149610741284
Social : 2.6627552747376737
Travel & Local : 2.324269434728647
Shopping : 2.245289405393208
Books & Reference : 2.1437436533904997
Simulation : 2.042197901387792
Dating : 1.8616721200496444
Arcade : 1.8503892587160102
Video Players & Editors : 1.771409229380571
Casual : 1.7601263680469368
Maps & Navigation : 1.399074805370642
Food & Drink : 1.241114746699763
Puzzle : 1.128286133363421
Racing : 0.9928917973598104
Role Playing : 0.9364774906916393
Libraries & Demo : 0.9364774906916393
Auto & Vehicles : 0.9251946293580051
S

## Data Analysis Part 2:

In the Apple Store we can see the most popular free English apps are:

1) Games : 58.1

2) Entertainment : 7.8

3) Photo & Video : 4.9

In the Google Play Store `category` we find:

1) FAMILY : 18.9

2) GAME : 9.7

3) TOOLS : 8.4

and in `genre` column of the Google Play Store:

1) Tools : 8.4

2) Entertainment : 6.0

3) Education : 5.3

## Data Analysis Part 3:

   For Apple, games are clearly the way to go. Overall, the top apps are in entertainment/games. It would be helpful to know within the `Games` group which ones are more successful?
   
   For Google Play, we find that more practical apps are most popular, with `Tools` being in the top three of both. `Game` and `Entertainment` both appear in the top Google apps, but are less likely a solid hit.
   
   It may be more helpful to also know the number of users within each genre. For GPS we can compare the `Installs`, but Apple doesn't have that. As a proxy we can compare the `rating_count_tot`

In [27]:
#iterating user counts in apples store data set

as_ft = freq_table(as_final, 11)

for genre in as_ft:
    total = 0
    len_genre = 0
    for app in as_final:
        genre_app = app[11]
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
            
    avr_rating = total/len_genre
    print(genre, ":", avr_rating)

Travel : 28243.8
Navigation : 86090.33333333333
Reference : 74942.11111111111
Medical : 612.0
Business : 7491.117647058823
Sports : 23008.898550724636
Shopping : 26919.690476190477
Photo & Video : 28441.54375
Music : 57326.530303030304
Entertainment : 14029.830708661417
Social Networking : 71548.34905660378
Food & Drink : 33333.92307692308
Utilities : 18684.456790123455
Finance : 31467.944444444445
Book : 39758.5
Productivity : 21028.410714285714
Catalogs : 4004.0
News : 21248.023255813954
Weather : 52279.892857142855
Games : 22788.6696905016
Health & Fitness : 23298.015384615384
Education : 7003.983050847458
Lifestyle : 16485.764705882353


For the Apple Store, the most popular (by number of user ratings) is `navigation` followed closely by `reference` and `social networking`

Lets dig into the Reference genre:

In [40]:
for app in as_final:
    if app[11] == 'Reference':
        print(app[1], ':', app[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


How about the Navigation genre:

In [42]:
for app in as_final:
    if app[11] == 'Navigation':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


A select few of very popular apps can pull and entire genre up. 

For Navigation, Waze is monsterously popular. In Reference the Bible app and Dictionary were very popular.

let's look at the `Installs` column for the Google Play Store

In [28]:
display_table(gps_final, 5)

1,000,000+ : 15.728308699086089
100,000+ : 11.55365000564143
10,000,000+ : 10.549475346947986
10,000+ : 10.188423784271691
1,000+ : 8.394448832223853
100+ : 6.916393997517771
5,000,000+ : 6.826131106848697
500,000+ : 5.562450637481666
50,000+ : 4.772650344127271
5,000+ : 4.513144533453684
10+ : 3.542818458761142
500+ : 3.2494640640866526
50,000,000+ : 2.3017037120613786
100,000,000+ : 2.1324607920568655
50+ : 1.9180864267178157
5+ : 0.7898002933543946
1+ : 0.5077287600135394
500,000,000+ : 0.270788672007221
1,000,000,000+ : 0.2256572266726842
0+ : 0.045131445334536835
0 : 0.011282861333634209


These totals are as precise as we would like, but should do in a pinch. Let's consider that 100,00+ installs is 100,000.

In [29]:
#iterating user counts in google play store data set

gps_ft = freq_table(gps_final, 1)

for category in gps_ft:
    total = 0
    len_category = 0
    for app in gps_final:
        category_app = app[1]
        if category_app == category:
            installs = app[5]
            installs = installs.replace("+", "")
            installs = installs.replace(",", "")
            total += float(installs)
            len_category += 1
            
    avr_installs = total/len_category
    print(category, ":", round(avr_installs, 2))

VIDEO_PLAYERS : 24727872.45
AUTO_AND_VEHICLES : 647317.82
BOOKS_AND_REFERENCE : 8767811.89
HEALTH_AND_FITNESS : 4188821.99
MAPS_AND_NAVIGATION : 4056941.77
COMICS : 817657.27
ART_AND_DESIGN : 2021626.79
LIFESTYLE : 1437816.27
TRAVEL_AND_LOCAL : 13984077.71
NEWS_AND_MAGAZINES : 9549178.47
PRODUCTIVITY : 16787331.34
PARENTING : 542603.62
FAMILY : 3695641.82
PHOTOGRAPHY : 17840110.4
TOOLS : 10801391.3
GAME : 15588015.6
DATING : 854028.83
SOCIAL : 23253652.13
FINANCE : 1387692.48
LIBRARIES_AND_DEMO : 638503.73
HOUSE_AND_HOME : 1331540.56
PERSONALIZATION : 5201482.61
ENTERTAINMENT : 11640705.88
MEDICAL : 120550.62
SPORTS : 3638640.14
BEAUTY : 513151.89
EDUCATION : 1833495.15
COMMUNICATION : 38456119.17
BUSINESS : 1712290.15
SHOPPING : 7036877.31
WEATHER : 5074486.2
FOOD_AND_DRINK : 1924897.74
EVENTS : 253542.22


Most installs in the Google Play Store include: `COMMUNICATION`, `PRODUCTIVITY`, `VIDEO PLAYER` 

Let's dive deeper into a few of those, starting with Communication:

In [30]:
for app in gps_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

Apps like What's App, Messenger, Skype skew the results heavily  with high downloads.

What about in Productivity?

In [32]:
for app in gps_final:
    if app[1] == 'PRODUCTIVITY' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Microsoft Word : 500,000,000+
Microsoft Outlook : 100,000,000+
Microsoft OneDrive : 100,000,000+
Microsoft OneNote : 100,000,000+
Google Keep : 100,000,000+
ES File Explorer File Manager : 100,000,000+
Dropbox : 500,000,000+
Google Docs : 100,000,000+
Microsoft PowerPoint : 100,000,000+
Samsung Notes : 100,000,000+
SwiftKey Keyboard : 100,000,000+
Google Drive : 1,000,000,000+
Adobe Acrobat Reader : 100,000,000+
Google Sheets : 100,000,000+
Microsoft Excel : 100,000,000+
WPS Office - Word, Docs, PDF, Note, Slide & Sheet : 100,000,000+
Google Slides : 100,000,000+
ColorNote Notepad Notes : 100,000,000+
Evernote – Organizer, Planner for Notes & Memos : 100,000,000+
Google Calendar : 500,000,000+
Cloud Print : 500,000,000+
CamScanner - Phone PDF Creator : 100,000,000+


The primary Office apps like Google Drive, Word, Google Calendar and Dropbox push the installs in the Producivity category.

## Interpretation and Recommendations

It would appear that Reference is a popular download in both stores. However, moany of these are mobile verions of popular PC/Apple programs.

Communications and/or Social Media is also popular, largely driven by a limited number of very popular apps.

Perhaps there is an opportunity for an app that connected a reference tool with communication. A way to better share documents, meda, etc?